cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/ApiController.java

```diff
@@ -13,18 +13,448 @@
  */
 package io.mesosphere.mesos.frameworks.cassandra;
 
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.google.common.base.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.StringWriter;
 
 @Path("/")
 public final class ApiController {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ApiController.class);
+
+    private final CassandraCluster cluster;
+
+    public ApiController(CassandraCluster cluster) {
+        this.cluster = cluster;
+    }
+
+    @GET
+    @Path("/")
+    @Produces("text/html")
+    public String indexPage() {
+        return "<a href=\"status\">All nodes</a> <br/>" +
+                "<a href=\"seed-nodes\">List of seed nodes</a> <br/>" +
+                "<a href=\"repair/start\">Start repair</a> <br/>" +
+                "<a href=\"repair/status\">Current repair status</a> <br/>" +
+                "<a href=\"repair/last\">Last repair</a> <br/>" +
+                "<a href=\"repair/abort\">Abort current repair</a> <br/>";
+    }
+
+    @GET
+    @Path("/seed-nodes")
+    @Produces("application/json")
+    public Response seedNodes() {
+        StringWriter sw = new StringWriter();
+        try {
+            CassandraFrameworkProtos.CassandraFrameworkConfiguration configuration = cluster.getConfiguration().get();
+
+            JsonFactory factory = new JsonFactory();
+            JsonGenerator json = factory.createGenerator(sw);
+            json.setPrettyPrinter(new DefaultPrettyPrinter());
+            json.writeStartObject();
+
+            json.writeNumberField("native_port", CassandraCluster.getPortMapping(configuration, CassandraCluster.PORT_NATIVE));
+            json.writeNumberField("rpc_port", CassandraCluster.getPortMapping(configuration, CassandraCluster.PORT_RPC));
+
+            writeSeedIps(json);
+
+            json.writeEndObject();
+            json.close();
+        } catch (Exception e) {
+            LOGGER.error("Failed to build seed list", e);
+            return Response.serverError().build();
+        }
+
+        return Response.ok(sw.toString(), "application/json").build();
+    }
+
+    private void writeSeedIps(JsonGenerator json) throws IOException {
+        json.writeArrayFieldStart("seeds");
+        for (String seed : cluster.getSeedNodes())
+            json.writeString(seed);
+        json.writeEndArray();
+    }
+
+    @GET
+    @Path("/status")
+    @Produces("application/json")
+    public Response status() {
```

Comment:
Actually I just tried return a protobuf message directly and it errored out with a cycle, so that answers that part of the question.  But why did you change the structure rather than something like:
```json
{
  "config": {
    "frameworkId": "",
    "frameworkName": "",
    "cassandraVersion": "",
    "numberOfNodes": "",
    "cpuCores": "",
    "memMb": "",
    "diskMb": "",
    "healthCheckIntervalSeconds": "",
    "numberOfSeeds": "",
    "bootstrapGraceTimeSeconds": "",
    "portMapping": [...]
  },
  "clusterState": {
    "nodes": [...],
    "executorMetadata": [...],
    "lastServerLaunchTimestamp": 123123123123
  }
}
```